### ISSUE TITLE
Spread element does not emit nullability warning when converting to certain collection types

### ISSUE THREAD

**Issue by just-ero** [dotnet/roslyn#78307]
**Full issue information + repro project**:  
https://github.com/just-ero/roslyn-issues-repros/tree/main/5.0.0/001_SpreadElement_NullabilityWarning

### Version

**Roslyn**: 5.0.0-1.25204.1 (ad1c386d)

### Steps to Reproduce

1. Add the following code wherever syntactically applicable:
    ```cs
    List<string> Repro()
    {
        string?[] arr = [];
        return [.. arr];
    }
    ```

### Expected Behavior

`[.. arr]` emits `CS8601` (`Possible null reference assignment.`).

### Actual Behavior

It does not.  
This is the case for `List<T>` and `HashSet<T>`. Other collection types may also be affected.

When the target collection type is `T[]` or `IEnumerable<T>`, the warning is produced as expected.

**Comment by RikkiGibson** [comment]
Huh, I thought I did not see a warning when target type was `T[]`. Possibly this issue is related to #68786, though I'm uncertain.

**Comment by cston** [comment]
> Related to [#68786](https://github.com/dotnet/roslyn/issues/68786). Possibly related to specific line from that issue:
> 
> > Checking conversion of elements to iteration type for collection initializer types

Yes, that is the work item.
